diff -ur fritz/src/tools.h fritz.new/src/tools.h
--- fritz/src/tools.h	2005-07-07 00:00:00.000000000 +0200
+++ fritz.new/src/tools.h	2006-01-26 17:29:53.000000000 +0100
@@ -71,13 +71,14 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
+#ifndef atomic_xchg
 static inline unsigned long atomic_xchg (
 	volatile atomic_t *	v, 
 	unsigned		value
 ) {
 	return __xchg (value, &v->counter, sizeof (unsigned));
 } /* atomic_xchg */
-
+#endif
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 #ifndef NDEBUG
diff -urp fritz/src.org/main.c fritz/src/main.c
--- fritz/src.org/main.c	2007-09-29 18:44:31.000000000 +0000
+++ fritz/src/main.c	2007-09-30 12:06:45.000000000 +0000
@@ -22,7 +22,6 @@
 
 #include <asm/uaccess.h>
 #include <linux/pci.h>
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
diff -urp fritz/src.org/driver.h fritz/src/driver.h
--- fritz/src.org/driver.h	2007-09-29 18:44:31.000000000 +0000
+++ fritz/src/driver.h	2007-09-30 11:54:32.000000000 +0000
@@ -23,7 +23,6 @@
 #define __have_driver_h__
 
 #include <asm/atomic.h>
-#include <linux/config.h>
 #include <linux/skbuff.h>
 #include <linux/pci.h>
 #include <linux/spinlock.h>
diff -urp fritz/src.org/main.c fritz/src/main.c
--- fritz/src.org/main.c	2007-09-29 18:44:31.000000000 +0000
+++ fritz/src/main.c	2007-09-30 12:06:45.000000000 +0000
@@ -47,6 +46,8 @@
 #include "lib.h"
 #include "defs.h"
 
+#define pci_module_init	pci_register_driver
+
 MODULE_LICENSE ("Proprietary");
 MODULE_DESCRIPTION ("CAPI4Linux: Driver for " PRODUCT_LOGO);
 
--- fritz/src/driver.c.orig	2005-07-06 22:00:00.000000000 +0000
+++ fritz/src/driver.c	2007-09-30 13:39:52.000000000 +0000
@@ -487,7 +487,7 @@ int start (card_p cp) {
 	res = request_irq (
 		cp->irq, 
 		&device_interrupt, 
-		SA_INTERRUPT | SA_SHIRQ, 
+		IRQF_DISABLED | IRQF_SHARED, 
 		TARGET, 
 		cp
 	);
--- fritz/src/devif.h~	2007-09-30 13:41:27.000000000 +0000
+++ fritz/src/devif.h	2007-09-30 13:41:27.000000000 +0000
@@ -80,7 +80,7 @@ extern __attr void dif_xfer_requirements
 extern void set_interrupt_callback (irq_callback_t, void *);
 extern void clear_interrupt_callback (void);
 
-extern irqreturn_t device_interrupt (int, void *, struct pt_regs *);
+extern irqreturn_t device_interrupt (int, void *);
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
--- fritz/src/devif.c~	2007-09-30 13:41:11.000000000 +0000
+++ fritz/src/devif.c	2007-09-30 13:41:11.000000000 +0000
@@ -747,12 +747,11 @@ static void xfer_task (unsigned long dat
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-irqreturn_t device_interrupt (int irq, void * args, struct pt_regs * regs) {
+irqreturn_t device_interrupt (int irq, void * args) {
 	unsigned long	intpins;
 	card_p		cp	= (card_p) args;
 
 	UNUSED_ARG (irq);
-	UNUSED_ARG (regs);
 	assert (capi_card == cp);
 
 	intpins = PINL (cp->mmio_base + C6205_PCI_HSR_OFFSET);
diff -urp fritz/src.org/driver.h fritz/src/driver.h
--- fritz/src.org/driver.h	2007-09-29 18:44:31.000000000 +0000
+++ fritz/src/driver.h	2007-09-30 11:54:32.000000000 +0000
@@ -98,7 +97,7 @@ extern int fcpcmcia_delcard (unsigned, u
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-extern int driver_init (void);
+extern int avm_driver_init (void);
 extern void driver_exit (void);
 
 /*---------------------------------------------------------------------------*\
--- fritz/src/driver.c.orig	2007-09-30 13:16:12.000000000 +0000
+++ fritz/src/driver.c	2007-09-30 13:18:16.000000000 +0000
@@ -1796,10 +1796,10 @@ void __stack init (unsigned len, void (_
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-int driver_init (void) {
+int avm_driver_init (void) {
 
 	return (NULL != (capi_lib = link_library (&capi_card)));
-} /* driver_init */
+} /* avm_driver_init */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
--- fritz/src/main.c.orig	2007-09-30 13:43:28.000000000 +0000
+++ fritz/src/main.c	2007-09-30 13:45:45.000000000 +0000
@@ -117,7 +117,7 @@ static int __devinit fcdsl2_probe (
 		return -ENODEV;
 	}
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!avm_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		NOTE("Not loaded.\n");
 		return -ENOSYS;
diff -ur fritz/src/driver.c.orig fritz/src/driver.c
--- fritz/src/driver.c.orig	2008-07-21 13:47:39.000000000 +0200
+++ fritz/src/driver.c	2008-07-21 13:48:55.000000000 +0200
@@ -116,7 +116,8 @@
 static DECLARE_WAIT_QUEUE_HEAD(capi_wait);
 static DECLARE_WAIT_QUEUE_HEAD(dbg_wait);
 
-static DECLARE_MUTEX_LOCKED(thread_sync);
+//static DECLARE_MUTEX_LOCKED(thread_sync);
+static DECLARE_COMPLETION(thread_sync);
 
 #define SCHED_WAKEUP_CAPI       { atomic_set (&thread_capi_flag, 1); wake_up_interruptible (&capi_wait); }
 #define SCHED_WAKEUP            { atomic_set (&got_kicked, 1); wake_up_interruptible (&wait); }
@@ -1405,7 +1406,8 @@
 		}
 	}
 	LOG("Scheduler thread stopped.\n");
-	up (&thread_sync);
+	//up (&thread_sync);
+	complete(&thread_sync);
 	return 0;
 } /* sched_thread */
 
@@ -1439,7 +1441,8 @@
 			SCHED_WAKEUP;
 		}
 		LOG("Thread signalled, waiting for termination...\n");
-		down (&thread_sync);
+		//down (&thread_sync);
+		wait_for_completion(&thread_sync);
 		LOG("Thread[%d] terminated.\n", thread_pid);
 	}
 	thread_pid = -1;
diff -ur fritz/src/common.h.orig fritz/src/common.h
--- fritz/src/common.h.orig	2008-07-21 13:46:41.000000000 +0200
+++ fritz/src/common.h	2008-07-21 13:46:54.000000000 +0200
@@ -44,7 +44,7 @@
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 typedef	long			intptr_t;
-typedef	unsigned long		uintptr_t;
+//typedef	unsigned long		uintptr_t;
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
--- fritz/src/driver.c	2012-02-04 18:04:25.000000000 +0100
+++ fritz/src/driver.c	2012-02-04 18:05:24.000000000 +0100
@@ -412,14 +412,14 @@
 static void exit_debug_buffer (dbg_buf_p dbp) {
 	
 	assert (dbp != NULL);
-	info (find_task_by_pid (dbp->pid) != NULL);
+//	info (find_task_by_pid (dbp->pid) != NULL);
 	LOG("Stopping debug thread...\n");
 	atomic_set (&dbp->stop, 1);
-	if (find_task_by_pid (dbp->pid)) {
+//	if (find_task_by_pid (dbp->pid)) {
 		wake_up_interruptible (&dbg_wait);
-	} else {
-		hfree (dbp);
-	}
+//	} else {
+//		hfree (dbp);
+//	}
 } /* exit_debug_buffer */
 
 /*---------------------------------------------------------------------------*\
@@ -1432,9 +1432,9 @@
 static void kill_thread (void) {
 
 	atomic_set (&thread_flag, 0);
-	if (NULL == find_task_by_pid (thread_pid)) {
-		LOG("Thread[%d] has died before!\n", thread_pid);
-	} else {
+//	if (NULL == find_task_by_pid (thread_pid)) {
+//		LOG("Thread[%d] has died before!\n", thread_pid);
+//	} else {
 		if (!atomic_read (&thread_capi_flag)) {
 			SCHED_WAKEUP_CAPI;
 		} else {
@@ -1444,7 +1444,7 @@
 		//down (&thread_sync);
 		wait_for_completion(&thread_sync);
 		LOG("Thread[%d] terminated.\n", thread_pid);
-	}
+//	}
 	thread_pid = -1;
 } /* kill_thread */
 
