--- linux-source-2.6.32-orig/scripts/mod/file2alias.c	2009-12-03 04:51:21.000000000 +0100
+++ linux-source-2.6.32/scripts/mod/file2alias.c	2009-12-18 20:20:57.000000000 +0100
@@ -727,6 +727,19 @@ static int do_platform_entry(const char 
 	return 1;
 }
 
+/* looks like: "pnp:dD" */
+static int do_isapnp_entry(const char *filename,
+			   struct isapnp_device_id *id, char *alias)
+{
+	sprintf(alias, "pnp:d%c%c%c%x%x%x%x*",
+		'A' + ((id->vendor >> 2) & 0x3f) - 1,
+		'A' + (((id->vendor & 3) << 3) | ((id->vendor >> 13) & 7)) - 1,
+		'A' + ((id->vendor >> 8) & 0x1f) - 1,
+		(id->function >> 4) & 0x0f, id->function & 0x0f,
+		(id->function >> 12) & 0x0f, (id->function >> 8) & 0x0f);
+	return 1;
+}
+
 /* Ignore any prefix, eg. some architectures prepend _ */
 static inline int sym_is(const char *symbol, const char *name)
 {
@@ -874,6 +887,10 @@ void handle_moddevtable(struct module *m
 		do_table(symval, sym->st_size,
 			 sizeof(struct platform_device_id), "platform",
 			 do_platform_entry, mod);
+	else if (sym_is(symname, "__mod_isapnp_device_table"))
+		do_table(symval, sym->st_size,
+			sizeof(struct isapnp_device_id), "isa",
+			do_isapnp_entry, mod);
 	free(zeros);
 }
 
--- linux-source-2.6.32-orig/drivers/isdn/hisax/hisax_fcpcipnp.c	2009-12-03 04:51:21.000000000 +0100
+++ linux-source-2.6.32/drivers/isdn/hisax/hisax_fcpcipnp.c	2009-12-17 23:55:08.000000000 +0100
@@ -74,9 +74,10 @@ static struct pnp_device_id fcpnp_ids[] 
 		.id		= "AVM0900",
 		.driver_data	= (unsigned long) "Fritz!Card PnP",
 	},
+	{ .id = "" }
 };
 
-MODULE_DEVICE_TABLE(isapnp, fcpnp_ids);
+MODULE_DEVICE_TABLE(pnp, fcpnp_ids);
 #endif
 
 static int protocol = 2;       /* EURO-ISDN Default */
