--- a/Makefile	2014-04-01 07:29:46.245164288 +0200
+++ b/Makefile	2014-04-01 07:39:18.000000000 +0200
@@ -54,10 +54,10 @@
 	$(CC) -c $(CFLAGS) -fno-strict-aliasing reloc.c
 
 test.o: test.c
-	$(CC) -c -Wall -march=i486 -m32 -O0 -fomit-frame-pointer -fno-builtin -ffreestanding test.c
+	$(CC) -c -Wall -march=i486 -m32 -Os -fomit-frame-pointer -fno-builtin -fno-pic -fno-pie -ffreestanding -fno-stack-protector test.c
 
 random.o: random.c
-	$(CC) -c -Wall -march=i486 -m32 -O3 -fomit-frame-pointer -fno-builtin -ffreestanding random.c
+	$(CC) -c -Wall -march=i486 -m32 -O3 -fomit-frame-pointer -fno-builtin -ffreestanding -fno-stack-protector random.c
 	
 # rule for build number generation  
 build_number:
